src/gui/Controls.cpp

```diff
@@ -38,7 +38,11 @@ namespace lmms::gui
 {
 
 
-void KnobControl::setText(const QString &text) { m_knob->setLabel(text); }
+void KnobControl::setText(const QString& text)
+{
+	// For KnobControls the text is set in the constructor
+	// so we do nothing here
+}
```

Comment:
@michaelgregorius Is there a reason for this discrepancy? Why can't `Knob::setLabel()` be public rather than protected?

Reply:
The idea is to prevent lots of code that looks as follows:
```
auto knob = new Knob(...);
knob->setLabel("foo");
```

If you search the diff of this PR for `setLabel` you will find that there were lots and lots of these "noisy" calls.

The text of a knob is one of its defining properties, i.e. it can for example be seen directly in the user interface. Hence it should be provided in the constructor. Technically the setter could of course be made public, and one might argue that it should be for something what I called a "defining property" some lines ago, but so far it does not really seem to be necessary.

Reply:
I can see it being temporarily useful for finding and fixing instances of those "noisy" calls, but now that they are converted to set the label via the constructor, I think `setLabel` should be public again.

Benefits of doing this:
- The label can be changed without needing to construct a new Knob
- The incorrect behavior in `KnobControl::setText` can be fixed
- The related method `Knob::setHtmlText` is not the only one that is public